parse.py

```diff
@@ -1046,7 +1046,7 @@ def signed(value, width):
     return value - (1<<width)
 
 
-if __name__ == "__main__":
+def main():
```

Comment:
@aswaterman yes, we can do this change as I have already did that in Refactored `parse.py` file in my PR ( a longer one). So yeah, this could be accepted.

Reply:
Let's close this PR and handle this as part of merging your PR.

Reply:
@IIITM-Jay which PR are you referring to? I can't find one that does this.

Reply:
@Timmmm  I have revised the PR #283 with mentioning in comments, about only to consider that PR with pure intention to refactor and optimize the parser logic with modular approach to enhance maintainability and readability keeping the best practices and standards.

Will include this thing as I proceed further refactoring other scripts which are dedicated to generate the corresponding artifacts such `encoding.out.h`, `instr-table.tex` etc. 
I have in mind to include the changes w.r.t `main` method inclusion.